pkg/collector/corechecks/cluster/orchestrator/transformers/ecs/task.go

```diff
@@ -0,0 +1,239 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build orchestrator
+
+// Package ecs provides methods for converting ECS resources to protobuf model.
+package ecs
+
+import (
+	"fmt"
+	"hash/fnv"
+	"sort"
+	"strconv"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+
+	model "github.com/DataDog/agent-payload/v5/process"
+	"github.com/DataDog/datadog-agent/comp/core/tagger"
+	"github.com/DataDog/datadog-agent/comp/core/tagger/collectors"
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/orchestrator"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var json = jsoniter.ConfigCompatibleWithStandardLibrary
+
+// TaskWithContainers represents an ECS task with its containers fetched from the workloadmeta store
+type TaskWithContainers struct {
+	Task       *workloadmeta.ECSTask
+	Containers []*workloadmeta.Container
+}
+
+// ExtractECSTask returns the protobuf model corresponding to an ECS Task resource.
+func ExtractECSTask(task TaskWithContainers) *model.ECSTask {
+	if task.Task == nil {
+		return nil
+	}
+	taskModel := &model.ECSTask{
+		Arn:                     task.Task.EntityID.ID,
+		LaunchType:              string(task.Task.LaunchType),
+		DesiredStatus:           task.Task.DesiredStatus,
+		KnownStatus:             task.Task.KnownStatus,
+		Family:                  task.Task.Family,
+		Version:                 task.Task.Version,
+		AvailabilityZone:        task.Task.AvailabilityZone,
+		Limits:                  task.Task.Limits,
+		EphemeralStorageMetrics: task.Task.EphemeralStorageMetrics,
+		ServiceName:             task.Task.ServiceName,
+		VpcId:                   task.Task.VPCID,
+		PullStartedAt:           extractTimestampPtr(task.Task.PullStartedAt),
+		PullStoppedAt:           extractTimestampPtr(task.Task.PullStoppedAt),
+		ExecutionStoppedAt:      extractTimestampPtr(task.Task.ExecutionStoppedAt),
+		Containers:              extractECSContainer(task.Containers),
+	}
+
+	tags, err := tagger.Tag(fmt.Sprintf("ecs_task://%s", task.Task.EntityID.ID), collectors.HighCardinality)
+	if err != nil {
+		log.Debugf("Could not retrieve tags for task: %s", err.Error())
+	}
+
+	taskModel.Tags = tags
+	taskModel.EcsTags = toTags(task.Task.Tags)
```

Comment:
Is there any normalization needed for those tags?

Reply:
I'm not sure if we should do the normalisation here as the tag standard of ecs and datadog are different, changing tags can confuse user. We can probably leave it as it is for now